lib/HLSL/DxilValidation.cpp

```diff
@@ -2345,7 +2345,83 @@ static void ValidateDxilOperationCallInProfile(CallInst *CI,
           CI, ValidationRule::SmOpcodeInInvalidFunction,
           {"CreateHandle", "Shader model 6.5 and below"});
     }
+  } break;
+    
+  // this intrinsic is generated when the SV_DispatchThreadID system value is used
+  case DXIL::OpCode::ThreadId: {  
+    // get the node launch type
+    DxilModule &DM = ValCtx.DxilMod;
+    if (!DM.HasDxilEntryProps(F)) {
+      break;
+    }
+
+    DxilEntryProps &entryProps = DM.GetDxilEntryProps(F);
+    DXIL::NodeLaunchType NLT = entryProps.props.Node.LaunchType;
+    if (NLT == DXIL::NodeLaunchType::Thread || NLT == DXIL::NodeLaunchType::Coalescing) {
+			ValCtx.EmitInstrFormatError(CI, ValidationRule::InstrInvalidSVInFunction,
+          {"ThreadId", NLT ==
+              DXIL::NodeLaunchType::Thread
+          ? "Thread"
+          : "Coalescing"});
+		}
+	} break;                
+  // this intrinsic is generated when the SV_GroupId
+  // system value is used
+  
+	case DXIL::OpCode::GroupId: {  
```

Comment:
nits: tab used for indentation here, and the comment above looks a little out of place.